packages/api/src/brevo-contact/dto/brevo-contacts.args.ts

```diff
@@ -1,12 +1,36 @@
 import { OffsetBasedPaginationArgs } from "@comet/cms-api";
-import { ArgsType, Field } from "@nestjs/graphql";
-import { IsOptional } from "class-validator";
+import { Type } from "@nestjs/common";
+import { ArgsType, Field, ID } from "@nestjs/graphql";
+import { Type as TransformerType } from "class-transformer";
+import { IsOptional, IsString, ValidateNested } from "class-validator";
 
-@ArgsType()
-export class BrevoContactsArgs extends OffsetBasedPaginationArgs {
-    // TODO: add scope
+import { EmailCampaignScopeInterface } from "../../types";
 
-    @Field(() => String, { nullable: true })
-    @IsOptional()
+export interface BrevoContactsArgsInterface extends OffsetBasedPaginationArgs {
+    scope: EmailCampaignScopeInterface;
+    targetGroupId?: string;
     email?: string;
 }
+
+export class BrevoContactsArgsFactory {
+    static create({ Scope }: { Scope: Type<EmailCampaignScopeInterface> }) {
+        @ArgsType()
+        class BrevoContactsArgs extends OffsetBasedPaginationArgs {
+            @Field(() => ID, { nullable: true })
+            @IsString()
+            @IsOptional()
+            targetGroupId?: string;
+
+            @Field(() => String, { nullable: true })
+            @IsOptional()
```

Comment:
```suggestion
            @IsOptional()
            @IsEmail()
```

Reply:
Added in [aef2ba9](https://github.com/vivid-planet/comet-brevo-module/pull/17/commits/aef2ba95f335c49529309dcd33be2aedd2bb0ef1)